frontend/src/metabase/admin/settings/components/SettingsGoogleForm/SettingsGoogleForm.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useRef } from "react";
+import { connect } from "react-redux";
+import { jt, t } from "ttag";
+import MetabaseSettings from "metabase/lib/settings";
+import ActionButton from "metabase/components/ActionButton";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import Breadcrumbs from "metabase/components/Breadcrumbs";
+import {
+  FormField,
+  FormMessage,
+  FormSubmit,
+} from "metabase/containers/FormikForm";
+import { updateGoogleSettings } from "metabase/admin/settings/settings";
+import { settingToFormField } from "metabase/admin/settings/utils";
+import {
+  FormCaption,
+  FormSection,
+  FormHeader,
+  FormRoot,
+} from "./SettingsGoogleForm.styled";
+
+export interface SettingElement {
+  key: string;
+}
+
+export interface SettingsGoogleFormProps {
+  elements?: SettingElement[];
+  settingValues?: Record<string, unknown>;
+  onSubmit: (settingValues: Record<string, unknown>) => void;
+}
+
+const SettingsGoogleForm = ({
+  elements = [],
+  settingValues = {},
+  onSubmit,
+}: SettingsGoogleFormProps) => {
+  const isEnabled = Boolean(settingValues["google-auth-enabled"]);
+  const isEnabledRef = useRef(isEnabled);
+
+  const handleSubmit = (values: Record<string, unknown>) => {
+    return onSubmit({ ...values, "google-auth-enabled": isEnabledRef.current });
+  };
+
+  const handleSaveAndEnableClick = (handleSubmit: () => void) => {
+    isEnabledRef.current = true;
+    return handleSubmit();
+  };
+
+  const handleSaveAndNotEnableClick = (handleSubmit: () => void) => {
+    isEnabledRef.current = false;
+    return handleSubmit();
+  };
+
+  return (
+    <FormRoot
+      initialValues={settingValues}
+      renderSubmit={({ canSubmit, handleSubmit }) => (
+        <>
+          <ActionButton
+            actionFn={() => handleSaveAndEnableClick(handleSubmit)}
+            primary={canSubmit}
+            disabled={!canSubmit}
+            normalText={isEnabled ? t`Save changes` : t`Save and enable`}
+            successText={t`Changes saved!`}
+          />
+          {!isEnabled && (
+            <ActionButton
+              actionFn={() => handleSaveAndNotEnableClick(handleSubmit)}
+              disabled={!canSubmit}
+              normalText={t`Save but don't enable`}
+              successText={t`Changes saved!`}
+            />
+          )}
+        </>
+      )}
+      disablePristineSubmit
+      overwriteOnInitialValuesChange
+      onSubmit={handleSubmit}
+    >
+      <Breadcrumbs
+        crumbs={[
+          [t`Authentication`, "/admin/settings/authentication"],
```

Comment:
Using urls like this leads to broken links when site-url is like `https://company.com/another-segment/`

Reply:
Also this is not changed. The second argument is passed to `Link` from the `react-router`. I think it should work correctly but haven't actually checked. I mean, this is how all our urls are passed to `Link`